profiles/common/modules/features/nexteuropa_newsroom/classes/NewsroomHelper.inc

```diff
@@ -0,0 +1,633 @@
+   *   Field data.
+   */
+  public static function getNodesAssociatedToTopics($topic_ids) {
+    $topic_ids = is_array($topic_ids) ? $topic_ids : array($topic_ids);
+    $cache_key = 'get_topic_entities' . md5(implode('', $topic_ids));
+    $cache = cache_get($cache_key, NEXTEUROPA_NEWSROOM_CACHE_TABLE);
+    if ($cache) {
+      $data = $cache->data;
+    }
+    else {
+      $query = db_select('field_data_field_associated_newsroom_topic', 'f')
+        ->fields('f')
+        ->condition('f.field_associated_newsroom_topic_tid', $topic_ids)
+        ->execute();
+      $data = NewsroomHelper::convertPdoToArray($query);
+
+      cache_set($cache_key, $data, NEXTEUROPA_NEWSROOM_CACHE_TABLE);
+    }
+    return $data;
+  }
+
+  /**
+   * Get combined node newsroom topics.
+   *
+   * @param object $node_wrapper
+   *   Node wrapper.
+   *
+   * @return array
+   *   Newsroom topics.
+   */
+  public static function getNodeTopics($node_wrapper) {
+    $nid = $node_wrapper->__isset('nid') ? $node_wrapper->nid->value() : NULL;
+    $cache_key = 'get_node_topics:' . $nid;
+    $cache = cache_get($cache_key, NEXTEUROPA_NEWSROOM_CACHE_TABLE);
+    if ($cache) {
+      $topics = $cache->data;
+    }
+    else {
+      $primary_topics = $node_wrapper->__isset('field_primary_topic') ? $node_wrapper->field_primary_topic->value() : NULL;
+      if (!empty($primary_topics)) {
+        $primary_topics = !is_array($primary_topics) ? array($primary_topics) : $primary_topics;
+      }
+      else {
+        $primary_topics = array();
+      }
+
+      // Get regular topics.
+      $topics = $node_wrapper->__isset('field_newsroom_topics') ? $node_wrapper->field_newsroom_topics->value() : NULL;
+      if (!empty($topics)) {
+        $topics = !is_array($topics) ? array($topics) : $topics;
+      }
+      else {
+        $topics = array();
+      }
+
+      $topics = array_merge($primary_topics, $topics);
+
+      cache_set($cache_key, $topics, NEXTEUROPA_NEWSROOM_CACHE_TABLE);
+    }
+    return $topics;
+  }
+
+  /**
+   * Active menu for newsroom items.
+   *
+   * Returns wich menu item should be active for a newsroom item
+   * based on the assigned topics.
+   */
+  public static function fetchActiveMenu($topic_id) {
+    $cache_key = 'fetch_active_menu' . $topic_id;
+    $cache = cache_get($cache_key, NEXTEUROPA_NEWSROOM_CACHE_TABLE);
+    if ($cache) {
+      $path = $cache->data;
+    }
+    else {
+      $path = FALSE;
+
+      // Check if they are added to menu.
+      $menu_items = array();
+      // Fetch entities associated to topic using the special field.
+      $topic_entities = NewsroomHelper::getNodesAssociatedToTopics($topic_id);
+
+      if (count($topic_entities) > 0) {
+        foreach ($topic_entities as $entity) {
+          // Could use entity_uri() but we would have to load the full entity...
+          $menu_link_items = NewsroomHelper::getMenuByPath(str_replace('_', '/', $entity['entity_type']) . '/' . $entity['entity_id']);
```

Comment:
you're right, the function is called once with parameters of taxonomy terms - in this case the name of the function is a bit tricky because it's way too general one even it's called from one place with specific data. but if you see the function as generic as it shows itself the reported issue can be hardcoded :) i think it would be great if the function will have a nice, more specific name, isn't it?